app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,63 @@
+  <h2 class="recipe-title"><%=@recipe.name%></h2>
+  <div class="recipe-detail">
+    <div class="recipe-detail-left">
+    <p><strong>Preparation time: <%=pluralize(@recipe.preparation_time, 'hour')%> </strong></p>
+    <p><strong>Cooking time: <%=pluralize(@recipe.cooking_time, 'hour')%></strong></p>
+    <p><%= @recipe.description %></p>
+  </div>
+
+  <div class="recipe-detail-right">
+  <%= form_for @recipe, url: toggle_public_path(id: @recipe.id), method: :post do |f| %>
+  <div class="swith-btn">
+    <p><%= @recipe.public? ? 'Public' : 'Private' %></p>
+    <label class="switch">
+    <div><%= f.submit "" %></div>
+      <%= f.check_box :public, { checked: @recipe.public } %>
+      <span class="slider round"></span>
+    </label>
+  </div>
+  </div>
+<% end %>
+
+  </div>
+
+
+<div class="btn-container">
+ <div class="btn-left">
+  <%= link_to 'Generate shoping list', new_food_path(), class: "btn btn-right mb-2" %>
```

Comment:
- [ ] When I click this button to generate the shopping list, I am redirected to the create new foods page instead(`http://localhost:3000/foods/new`). Please fix this to make sure that a user generates a shopping list from the recipe ingredients provided :+1:

Reply:
Thank you for your feedback, it has now being added:

![Generate shopping list for single recipe](https://raw.githubusercontent.com/jorgeabrahan/recipe-app/d9e3a4b6e67a76d4b7af583b5079a0367034e856/repo_images/generate_shopping_list_for_single_recipe.png)